stream.go

```diff
@@ -2,19 +2,40 @@ package gtrs
 
 import (
 	"context"
+	"strconv"
+	"time"
 
 	"github.com/redis/go-redis/v9"
 )
 
+var NoExpiration = time.Duration(0)
+
+// now is defined here so it can be overridden in unit tests
+var now = time.Now
+
 // Stream represents a redis stream with messages of type T.
 type Stream[T any] struct {
 	client redis.Cmdable
 	stream string
+	ttl    time.Duration
+}
+
+type Options struct {
+	// TTL is an optional parameter to specify how long entries stay in the stream before expiring,
+	// it only only works as expected when a non-custom id is used to Add a message.
+	// The default is No Expiration.
+	// Note that TTL is performed when messages are Added, so Range requests won't clean up old messages.
+	TTL time.Duration
 }
 
 // Create a new stream with messages of type T.
-func NewStream[T any](client redis.Cmdable, stream string) Stream[T] {
-	return Stream[T]{client: client, stream: stream}
+// Options are optional (the parameter can be nil to use defaults).
+func NewStream[T any](client redis.Cmdable, stream string, opt *Options) Stream[T] {
+	ttl := NoExpiration
+	if opt != nil {
+		ttl = opt.TTL
+	}
```

Comment:
Not against the nil option. Libraries for example (go redis) usually expect a non nil pointer for all cases... But lets maybe keep it 🤷🏻 🙂

Reply:
Yeah trying to be defensive here and also make it slightly easier for callers who have no options to specify. I don't feel too strongly about it either, but it's already in there and used by the unit tests, so if you're OK with it, I am too 😉